database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -382,64 +404,93 @@ private String getSQLType(Class<?> fieldType) {
-        StringBuilder values = new StringBuilder();
+    private void setPreparedStatementValuesForInsert(PreparedStatement preparedStatement, BaseEntity entity)
+            throws SQLException {
+        setPreparedStatementValues(preparedStatement, entity, false, null);
+    }
+
+    private void setPreparedStatementValuesForUpdate(PreparedStatement preparedStatement, BaseEntity entity, Integer id)
+            throws SQLException {
+        setPreparedStatementValues(preparedStatement, entity, true, id);
+    }
 
+    private void setPreparedStatementValues(PreparedStatement preparedStatement, BaseEntity entity,
+                                            boolean includeId, Integer id) throws SQLException {
         List<Field> entityFields = getAllFields(entity.getClass());
-        for (Field field : entityFields) {
-            field.setAccessible(true);
+        int parameterIndex = 1;
 
+        for (Field field : entityFields) {
             if (field.getName().equals(ID_PARAMETER_NAME)) {
                 continue;
             }
-
-            fields.append(field.getName()).append(", ");
+            field.setAccessible(true);
             try {
                 Object value = field.get(entity);
 
-                if (value instanceof String || value instanceof Date) {
-                    values.append("'").append(value).append("', ");
+                if (value != null) {
+                    if (value instanceof String) {
+                        preparedStatement.setString(parameterIndex, (String) value);
+                    } else if (value instanceof Integer) {
+                        preparedStatement.setInt(parameterIndex, (Integer) value);
+                    } else if (value instanceof Long) {
+                        preparedStatement.setLong(parameterIndex, (Long) value);
+                    } else if (value instanceof Date) {
+                        preparedStatement.setDate(parameterIndex, new java.sql.Date(((Date) value).getTime()));
+                    } else if (value instanceof Boolean) {
+                        preparedStatement.setBoolean(parameterIndex, (Boolean) value);
+                    } else {
+                        preparedStatement.setObject(parameterIndex, value);
+                    }
                 } else {
-                    values.append(value).append(", ");
+                    preparedStatement.setNull(parameterIndex, java.sql.Types.NULL);
                 }
-            } catch (IllegalAccessException e) {
-                throw new RuntimeException("Error accessing field value", e);
+                parameterIndex++;
+            } catch (IllegalAccessException | SQLException e) {
+                LOG.error("Failed to set prepared statement values for field: {}", field.getName(), e);
+                throw new RuntimeException("Error setting prepared statement values", e);
             }
         }
 
+        if (includeId && id != null) {
```

Comment:
You can cove this with common `try` block and avoid of throwing `SQLException` from method.